configuration/developing-community-health-applications.md

```diff
@@ -219,37 +219,35 @@ More complex tasks can be written using the full set of properties for tasks, as
 | `events` | `object[]` | An event is used to specify the timing of the task. | yes |
-| `events[n].id` | `string` | Can help as a descriptive name (eg `pregnancy-high-risk`). One task will appear per unique id, so re-using ids can be useful to avoid duplicate tasks appearing. | no |
+| `events[n].id` | `string` | A descriptive identifier. Used for querying task completeness. | yes, unique |
 | `events[n].days` | `integer` | Number of days after the doc's `reported_date` that the event is due | yes, if `dueDate` is not set |
 | `events[n].dueDate` | `function(event, contact, report)` | Returns a `Date` object for the day when this event is due. | yes, if `days` is not set |
 | `events[n].start` | `integer` | Number of days to show the task before it is due. | yes |
 | `events[n].end` | `integer` | Number of days to show the task after it is due. | yes |
 | `actions` | `object[]` | The actions (forms) that a user can access after clicking on a task. If you put multiple forms here, the user will see a task summary screen where they can select which action they would like to complete. | yes |
 | `actions[n].type` | `'report'` or `'contact'` | When `'report'`, the action opens the given form. When `'contact'`, the action redirects to a contact's profile page. Defaults to 'report'. | no |
 | `actions[n].form` | `string` | The code of the form that should open when you select the action. | yes |
-| `actions[n].label`| `translation key` or `translation array` | The label that should appear on the task summary screen if multiple actions are present. | no |
+| `actions[n].label`| `translation key` | The label that should appear on the task summary screen if multiple actions are present. | no |
 | `actions[n].modifyContent`| `function (content, contact, report)` | Set the values on the content object to control the data which will be passed as `inputs` to the form which opens when the action is selected. | no |
-| `priority` | `object` | Controls the "high risk" line seen above. | no |
+| `priority` | `object` or `function(contact, report)` returning object of same schema | Controls the "high risk" line seen above. | no |
 | `priority.level` | `high` or `medium` | Tasks that are `high` will display a high risk icon with the task. Default: `medium`. | no |
-| `priority.label` | `translation key` or `translation array` | Text shown with the task associated to the risk level. | no |
-
-### Additional code
-Helper variables and functions can be defined in `nools-extras.js` and will be visible through the variable `extras`. This helps to keep the task definitions easy to read and manage. To enable reuse of common code, `nools-extras.js` file is shared by both the Tasks and Targets.
+| `priority.label` | `translation key` | Text shown with the task associated to the risk level. | no |
 
 ### Examples
 
 #### tasks.js
 ```js
-var isFormFromArraySubmittedInWindow = extras.isFormFromArraySubmittedInWindow;
+const extras = require('./nools-extras');
```

Comment:
Below nools-extras.js changed to targets-extras.js, should we also update this to `tasks-extras`? Is it useful to show this as the same file as targets to encourage functions to be reused, or the opposite for efficiency?

Reply:
In general, putting all functions into a "shared module" isn't particularly effective at encouring sharing. Instead, it often makes shared modules cluttered and disorganized. It can become hard to find the most useful utilities.

When writing shared libraries, the conventional wisdom is to make things as tightly scoped as possible. In Muso what we do is -- functions that are used only by targets go in targets-extras.js. If functions are used by both targets and tasks, they go in nools-extras.js. If functions are used by nools (either targets/tasks) and contact-summaries, they go in common.js. We might recommend this as a best-practice, but I'm not sure how useful it is the control how partners make shared libraries -- as long as things are organized logically and cleanly.

Unsure what the best path forward is for this doc -- but open to discussion/ideas.